File: src/test/java/ar/edu/unq/unidad3/dao/helper/service/DataServiceImpl.java
package ar.edu.unq.unidad3.dao.helper.service;

import ar.edu.unq.unidad3.dao.helper.dao.DataDAO;
import ar.edu.unq.unidad3.service.runner.HibernateTransactionRunner;

public class DataServiceImpl implements DataService {

    private final DataDAO dataDAO;

    public DataServiceImpl(DataDAO dataDAO) {
        this.dataDAO = dataDAO;
    }

    @Override
    public void cleanAll() {
        HibernateTransactionRunner.runTrx(() -> {
            dataDAO.clear();
            return null;
        });
    }
}

File: src/test/kotlin/ar/edu/unq/unidad3/dao/helper/service/DataServiceImpl.kt
package ar.edu.unq.unidad3.dao.helper.service

import ar.edu.unq.unidad3.dao.PersonajeDAO
import ar.edu.unq.unidad3.dao.helper.dao.DataDAO
import ar.edu.unq.unidad3.modelo.Item
import ar.edu.unq.unidad3.modelo.Personaje
import ar.edu.unq.unidad3.service.runner.HibernateTransactionRunner.runTrx

class DataServiceImpl (
    private val dataDAO: DataDAO
) : DataService {

    override fun cleanAll() {
        runTrx {
            dataDAO.clear()
        }
    }

}
